### DISCUSSION TITLE
Making releases

### DISCUSSION THREAD

**Discussion by deeplook** [giswqs/leafmaptools#6] [General]
@giswqs Could you briefly describe the workflow for making new releases, as this is is based on your project template? Thanks in advance! ;)

**Comment by giswqs** [comment]
This [video](https://youtu.be/yVFDdI-0E1M) introduces the key steps to making a new release and publishing it on PyPI and conda-forge. I summarize the key steps below. You can give it a try. I would also be happy to do that. Once the new release is on PyPI, I can update the package feedstock on conda-forge. 

1. install the `bump2version` package if needed using `pip install bump2version` or `pip install -r requirements_dev.txt`
2. bump the package version using `bumpversion patch` (which increases version +0.0.1), or `bumpversion minor` (version +0.1.0), or `bumpversion major` (version +1.0.0) 
3. push changes to GitHub using `git push --tags` and `git push` 
4. create a new release on GitHub based on the tagged version, which will trigger GitHub Actions ([leafmaptools/.github/workflows/pypi.yml](https://github.com/giswqs/leafmaptools/blob/master/.github/workflows/pypi.yml)) to automatically deploy the release to PyPI.

**Reply by deeplook** [reply]
Cool, I've just rolled a release.

**Reply by deeplook** [reply]
Ssems like some more permission is needed to deploy to pypi.

**Reply by giswqs** [reply]
I just added PyPI env variables to GitHub. Problem solved. https://pypi.org/project/leafmaptools. Will update conda-forge shortly.

**Reply by giswqs** [reply]
Feedstock updated. The new release should appear on conda-forge shortly. 
https://github.com/conda-forge/leafmaptools-feedstock/pull/1
